applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClientBase.java

```diff
@@ -0,0 +1,327 @@
+package com.azure.applicationconfig;
+
+import com.azure.applicationconfig.models.ConfigurationSetting;
+import com.azure.applicationconfig.models.SettingFields;
+import com.azure.applicationconfig.models.SettingSelector;
+import com.azure.common.ServiceClient;
+import com.azure.common.exception.ServiceRequestException;
+import com.azure.common.http.HttpPipeline;
+import com.azure.common.http.rest.PagedResponse;
+import com.azure.common.http.rest.Response;
+import com.azure.common.implementation.RestProxy;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+abstract class ConfigurationClientBase<T, U> extends ServiceClient {
```

Comment:
Not a fan of this approach. You've now got public API that extends from non-public API. Even if this is made public, I'm not convinced that this approach is going to yield the best outcome.

Reply:
This is one of those places where you should probably favour composition over inheritance.

Reply:
Agreed, I will go ahead and change the implementation to that. I initially decided on the inheritance approach with the idea that if the logic diverged in the future it would be easier to split the clients apart.